File: src/test/java/org/fabiomsr/javaxml/_01_deserialization/DeserializationTest.java
package org.fabiomsr.javaxml._01_deserialization;


import org.fabiomsr.javaxml._00_serialization.model.Book;
import org.junit.Test;

import java.io.File;
import java.net.URL;

import static org.junit.Assert.assertTrue;

/**
 * Created by fabiomsr on 2/7/16.
 */
public class DeserializationTest {

    @Test
    public void deserializationTest() throws Exception {
        URL bookXmlUrl = DeserializationTest.class.getResource("/book1.xml");
        File bookXml = new File(bookXmlUrl.toURI());

        DeserializationExample serializationExample = new DeserializationExample();

        Book resultBook = serializationExample.deserializeBook(bookXml);
        Book book = new Book("A Song of Ice and Fire", 864);

        assertTrue(book.equals(resultBook));
    }
}

File: src/test/kotlin/org/fabiomsr/kotlinxml/_01_deserialization/DeserializationTest.kt
package org.fabiomsr.kotlinxml._00_serialization

import org.fabiomsr.kotlinxml._00_serialization.model.Book
import org.fabiomsr.kotlinxml._01_deserialization.deserializeBook
import org.junit.Test
import java.io.File
import org.junit.Assert.assertTrue

/**
 * Created by fabiomsr on 2/7/16.
 */
class DeserializationTest {

    @Test
    fun deserializationTest() {
        val bookXmlUrl = javaClass.getResource("/book1.xml")
        val bookXml = File(bookXmlUrl.toURI())

        val resultBook = deserializeBook(bookXml)
        val book = Book("A Song of Ice and Fire", 864)

        assertTrue(book.equals(resultBook))
    }
}
